src/main/java/microsoft/exchange/webservices/data/core/request/SingletonEwsExecutor.java

```diff
@@ -0,0 +1,48 @@
+package microsoft.exchange.webservices.data.core.request;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.annotation.PreDestroy;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class SingletonEwsExecutor {
+    private static SingletonEwsExecutor singletonEwsExecutor = null;
+
+    private final ExecutorService executorService;
+
+    private static final Log log = LogFactory.getLog(SingletonEwsExecutor.class);
+
+    private SingletonEwsExecutor() {
+      executorService = Executors.newCachedThreadPool();
+    }
+
+    public static SingletonEwsExecutor getInstance() {
```

Comment:
I suspect this is being used in a multi-threaded context, which means `getInstance()` has to be `synchronized`, or different threads could end up with different objects.

Reply:
A better way to do this is to initialize the static singleton at declaration time, then you won't need sync of any sort.

Reply:
`
private static SingleonEwsExecutor singletonEwsExecutor = new SingletonEwsExecutor();
`

Will sort you out, then your get just returns the singleton

Reply:
Moved constructor call to variable declaration to create it at runtime.